src/utils/utils.rs

```diff
@@ -627,12 +627,26 @@ where
 }
 
 pub fn delete_dir_contents(dir_path: &Path) {
-    match remove_dir_all::remove_dir_all(dir_path) {
-        Err(e) if e.kind() != io::ErrorKind::NotFound => {
-            panic!("Unable to clean up {}: {:?}", dir_path.display(), e);
+    // Deletes only the contents, and not the directory iteself (in
+    // case it's a mountpoint, or a symlink, or has unusual
+    // permissions which the user wants to preserve).
+    use anyhow::Context;
+    (|| {
+        let entries = match fs::read_dir(dir_path) {
+            Err(e) if e.kind() == io::ErrorKind::NotFound => return Ok(()),
+            r => r,
         }
-        _ => {}
-    }
+        .context("checking directory")?;
+        for entry in entries {
+            let entry_path = entry.context("reading directory")?.path();
+            remove_dir_all::remove_dir_all(&entry_path)
```

Comment:
rather than reading the dir here and recursing into remove_dir_all, please patch remove_dir_all directly; inspecting https://github.com/XAMPPRocky/remove_dir_all/blob/master/src/fs.rs#L36 will show you why.  I have commit and publish rights on it, so can ensure a good response time.

I'm slightly  in favour of a new method e.g. remove_dir_contents_all in that crate rather than a parameter to the existing function (but probably refactor the contents to use a common internal helper).